contracts/FlowTransactionScheduler.cdc

```diff
@@ -0,0 +1,1440 @@
+            let slotEfforts = self.slotUsedEffort[tx.scheduledTimestamp]!
+            slotEfforts[tx.priority] = slotEfforts[tx.priority]!.saturatingSubtract(tx.executionEffort)
+            self.slotUsedEffort[tx.scheduledTimestamp] = slotEfforts
+
+            let totalFees = tx.fees
+            let refundedFees <- tx.payAndRefundFees(refundMultiplier: self.configurationDetails.refundMultiplier)
+
+            // if the transaction was canceled, add it to the canceled transactions array
+            // maintain sorted order by inserting at the correct position
+            var insertIndex = 0
+            for i, canceledID in self.canceledTransactions {
+                if id < canceledID {
+                    insertIndex = i
+                    break
+                }
+                insertIndex = i + 1
+            }
+            self.canceledTransactions.insert(at: insertIndex, id)
+            
+            // keep the array under the limit
+            if UInt(self.canceledTransactions.length) > self.configurationDetails.canceledTransactionsLimit {
+                self.canceledTransactions.remove(at: 0)
+            }
+
+            emit Canceled(
+                id: tx.id,
+                priority: tx.priority.rawValue,
+                feesReturned: refundedFees.balance,
+                feesDeducted: totalFees - refundedFees.balance,
+                transactionHandlerOwner: tx.handler.address,
+                transactionHandlerTypeIdentifier: tx.handlerTypeIdentifier
+            )
+
+            self.removeTransaction(txData: tx)
+            
+            return <-refundedFees
+        }
+
+        /// execute transaction is a system function that is called by FVM to execute a transaction by ID.
+        /// The transaction must be found and in correct state or the function panics and this is a fatal error
+        ///
+        /// This function is only called by the FVM to execute transactions.
+        access(Execute) fun executeTransaction(id: UInt64) {
+            let tx = self.borrowTransaction(id: id) ?? 
+                panic("Invalid ID: Transaction with id \(id) not found")
+
+            assert (
+                tx.status == Status.Executed,
+                message: "Invalid ID: Cannot execute transaction with id \(id) because it has incorrect status \(tx.status.rawValue)"
+            )
+
+            let transactionHandler = tx.handler.borrow()
+                ?? panic("Invalid transaction handler: Could not borrow a reference to the transaction handler")
+
+            emit Executed(
+                id: tx.id,
+                priority: tx.priority.rawValue,
+                executionEffort: tx.executionEffort,
+                transactionHandlerOwner: tx.handler.address,
+                transactionHandlerTypeIdentifier: transactionHandler.getType().identifier
+            )
+            
+            transactionHandler.executeTransaction(id: id, data: tx.getData())
```

Comment:
nit: I feel with the current naming this should be just `execute()` but I don't feel strongly about it. I just think this is actually not transaction on the contract call level.

Reply:
okay, I can update this to `execute()`

Reply:
Actually, it seems `execute` is a keyword, so we can't use it